ion/symbols_table.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ion
+
+type symbolTableKind uint8
+
+const (
+	symbolTableKindSystem symbolTableKind = iota
+	symbolTableKindShared
+	symbolTableKindLocal
+)
+
+// SymbolTable is the core lookup structure for Tokens from their symbolic ID
+type SymbolTable struct {
+	kind    symbolTableKind
+	name    string
+	version int32
+	imports []SymbolTable
+	textMap map[string]SymbolToken
+	tokens  []SymbolToken
+	maxSID  int64
```

Comment:
There is a collection of useful linters called `golangci-lint`.  One of the included linters is called `maligned` and it tells you when you have a data structure where the members could be re-arranged to save space.  I'm not certain that it would want to re-arrange this, but it's very possible.

Reply:
Added a TODO.